cheroot/test/test_https.py

```diff
@@ -0,0 +1,146 @@
+"""HTTPS Tests."""
+# -*- coding: utf-8 -*-
+# vim: set fileencoding=utf-8 :
+
+import os
+import pytest
+import ssl
+
+from cheroot.ssl.builtin import BuiltinSSLAdapter
+from cheroot.test import helper
+import cheroot
+import ddt
+
+
+def create_wsgi_server(**conf):
+    """Create test server."""
+    ssl_adapter = BuiltinSSLAdapter(
+        certificate=conf.pop('certificate'),
+        private_key=conf.pop('private_key'),
+        certificate_chain=conf.pop('certificate_chain'))
+    ssl_adapter.context.verify_mode = conf.pop('verify_mode', ssl.CERT_NONE)
+    server = cheroot.wsgi.Server(**conf)
+    server.ssl_adapter = ssl_adapter
+    return server
+
+
+def ssl_file(filename):
+    """Return path to given ssl file."""
+    return os.path.join('cheroot/test/ssl/', filename)
+
+
+class HTTPSTestBase(object):
+    """Base class for HTTPS tests."""
+
+    available_servers = {'wsgi': create_wsgi_server}
+    config = {
+        'certificate': 'cheroot/test/ssl/server.cert',
+        'private_key': 'cheroot/test/ssl/server.key',
+        'certificate_chain': 'cheroot/test/ssl/ca.cert',
+    }
+    script_name = '/hello'
+    ssl_context = None
+
+    @classmethod
+    def setup_server(cls):
+        """Setup the test server."""
+        class Root(helper.Controller):
+
+            def hello(req, resp):
+                return 'Hello world!'
+
+            handlers = {'/hello': hello}
+
+        cls.httpserver.wsgi_app = Root()
+
+    def setUp(self):
+        """TestCase setup."""
+        super().setUp()
+        self.ssl_context = ssl.create_default_context(
+            cafile='cheroot/test/ssl/ca.cert')
+        self.ssl_context.check_hostname = False
+
+    def getPage(self, *args, **kw):
+        """Fetch the page."""
+        return super().getPage(self.script_name, *args,
```

Comment:
This is incompatible with Python 2.7 https://travis-ci.org/cherrypy/cheroot/jobs/336749016#L543